packages/visual-editor/src/components/puck/Breadcrumbs.tsx

```diff
@@ -6,11 +6,16 @@ export type BreadcrumbsProps = {
   separator?: string;
 };
 
+// getDirectoryParents returns an array of objects. If no dm_directoryParents or children of
+// the directory parent are not objects, returns an empty array.
 const getDirectoryParents = (
   document: Record<string, any>
 ): Array<{ slug: string; name: string }> => {
   for (const key in document) {
-    if (key.startsWith("dm_directoryParents_")) {
+    if (
+      key.startsWith("dm_directoryParents_") &&
+      typeof document[key]?.[0] === "object"
```

Comment:
We may as well verify that the object shape is what the component expects:
```
{
  name: string,
  slug: string,
}
```
as a list